praw/models/reddit/base.py

```diff
@@ -29,22 +39,35 @@ def fullname(self):
 
         """
         return '{}_{}'.format(self._reddit._objector.kind(self),
-                              self.id)  # pylint: disable=invalid-name
+                              self.id)
+
+    @property
+    def fetched(self):
+        """Return True if reddit attributes have been fetched."""
+        return self._fetched
 
     def __eq__(self, other):
         """Return whether the other instance equals the current."""
-        if isinstance(other, basestring):  # basestring needed for py2 support
+        if isinstance(other, string_types):
             return other.lower() == str(self).lower()
         return (isinstance(other, self.__class__) and
                 str(self).lower() == str(other).lower())
 
-    def __getattr__(self, attribute):
-        """Return the value of `attribute`."""
-        if not attribute.startswith('_') and not self._fetched:
+    def __ne__(self, other):
```

Comment:
I didn't make the connection that it was moved. I prefer to keep classes/methods/variables sorted where possible (the only exception is when the definition of a class/method depends on another class/method).

Reply:
I think that things should be written in alphabetical order only if it makes sense to do so, and not have a hard and fast rule about member order on classes.

For class members, don’t you think it makes more sense to group things that relate, e.g., grouping together `__lt__`, `__le__`, `__eq__`, `__ne__`, `__gt__`, `__ge__`? If `__eq__` and `__ne__` are always written together then it makes looking for either one easier because finding one means you’ve essentially found the other one too.

Did you want `__ne__` back where it was? I could see that it was in alphabetical order, but I don’t see much value in placing it back away from `__eq__` if it doesn’t make things read better.